src/main/java/io/vertx/core/http/impl/HttpClientRequestImpl.java

```diff
@@ -57,12 +60,13 @@
 
   private static final Logger log = LoggerFactory.getLogger(HttpClientRequestImpl.class);
 
-  private final String host;
-  private final int port;
+  private String host;
+  private int port;
+  private int redirectCount;
   private final HttpClientImpl client;
-  private final HttpRequest request;
+  private HttpRequest request;
```

Comment:
if request is not final anymore, access need to be guarded by the monitor lock. It's not the case in uri().